### ISSUE TITLE
Issue with Assets API

### ISSUE THREAD

**Issue by VOA77** [XeroAPI/Xero-NetStandard#363]
Hi guys

**SDK you're using (please complete the following information):**
 - Xero.NetStandard.OAuth2 3.16.0
 - Xero.NetStandard.OAuth2Client 1.3.0

**Describe the bug**
We have an issue with getting data from Xero Assets API, via SDK Xero.NetStandard.OAuth2.
We tried to fetch the assets (status='Registered') but the method AssetApi.GetAssetsAsync returns null value. 
Also, we've tried running the same query via Postman and it worked fine.
Could you please help us with this issue? Is there an issue on SDK side and can we expect it to be fixed in the near future?

**To Reproduce**
1. fetch the Assets with the AssetApi.GetAssetsAsync method call
`await api.GetAssetsAsync(
      token.AccessToken,
      _tenantId,
      status,
      pageNumber,
      100
  );`

**Expected behavior**
return value of AssetApi.GetAssetsAsync is not null

**Screenshots**
![sample](https://user-images.githubusercontent.com/86400238/123217611-6f929f00-d4d3-11eb-8f6d-be5a0319f411.png)
Do not pay attention to the Throttle here. It's only the extension method which calls 'Thread.Sleep'.

**Comment by VOA77** [comment]
Looks like it appears because of parsing error of EffectiveLifeYears value from BookDepreciationSetting.

**Comment by angryfossa** [comment]
Hi, I am having this same issue and was wondering if there was any fix?

So far using `await AssetApi.GetAssetsAsync` and status='REGISTERED' and status='DISPOSED' cause this issue for me, I can call status='DRAFT' with no problems.
I've used the API explorer for the queries and it returns a result just fine. I've checked my app's history in the developer's section and it reports 0 errors with the query.

I did test `await AssetApi.GetAssetsAsyncWithHttpInfo` to see if the same issue occurs. It half does; when trying to view the 'Data' of the response its null, but when looking at the 'Content' of the response it seems to contain the valid information.

Thank you for any extra help anyone can provide.

**Comment by ionican** [comment]
Hi,  I am having the same issue - querying with the required status='REGISTERED' or 'DISPOSED' (set through the AssetStatusQueryParam enum) results in no data:

assets = Await apiInstance.GetAssetsAsync(xeroToken, "", status:=Xero.NetStandard.OAuth2.Model.Asset.AssetStatusQueryParam.REGISTERED, page:=1)

Knowing that the API is case sensitive for 'status' - could there be an issue with the way enum is converted in the HTTPS request?  (Interestingly I remember this being an issue with the old .net api back in the day).

**Comment by ionican** [comment]
> Looks like it appears because of parsing error of EffectiveLifeYears value from BookDepreciationSetting.

This is correct.  EffectiveLifeYears is defined as int? in BookDepreciationSetting.cs when it looks like it should be decimal?

Can confirm having made the change to a local clone that the wrapper now returns the assets correctly.
